### PR TITLE
Update event schema to include content_encoding field

### PR DESCRIPTION
**Pull request by cheng-tan**
Adding in this field to determine using current mode or action dedup + zstd.

### CODE REVIEW

**Review by kumpera — CHANGES_REQUESTED** [review]
Making this an enum is problematic because we might decide down the line to pick just dedup or compression. It should be a string.

Additionally, we didn't change rlclientlib to send the default value of identity.

**Comment by cheng-tan** [comment]
> Making this an enum is problematic because we might decide down the line to pick just dedup or compression. It should be a string.
> 
> Additionally, we didn't change rlclientlib to send the default value of identity.

Will set the default value to identity.

Regards to enum vs string, if we only want to pick dedup or compression, can we just add another option in the enum? won't string be more error prone if we got some random string not supported?

**Review by kumpera — CHANGES_REQUESTED** [review]
Content encoding should be a string.
This PR needs tests that we're encoding the right value.

File: rlclientlib/serialization/fb_serializer.h [hunk 1]
```diff
@@ -273,6 +275,59 @@ namespace reinforcement_learning { namespace logger {
...
+      _event_offsets.push_back(offset);
+      return error_code::success;
+    }
+
+    uint64_t size() const { return _builder.GetSize(); }
+
+    const char * getTextForEnum(content_encoding_enum enumVal)
+    {
+      if (enumVal == content_encoding_enum::ZSTD_AND_DEDUP) {
+        return "ZSTD_AND_DEDUP";
```
Comment (kumpera) [inline rlclientlib/serialization/fb_serializer.h:301]: We should use the config constants for those two.

File: rlclientlib/serialization/fb_serializer.h [hunk 2]
```diff
@@ -273,6 +275,59 @@ namespace reinforcement_learning { namespace logger {
...
+    {
+      flatbuffers::Offset<typename serializer_t::fb_event_t> offset;
+      RETURN_IF_FAIL(serializer_t::serialize(evt, _builder, offset, status));
+      _event_offsets.push_back(offset);
+      return error_code::success;
+    }
+
+    uint64_t size() const { return _builder.GetSize(); }
+
+    const char * getTextForEnum(content_encoding_enum enumVal)
```
Comment (kumpera) [inline rlclientlib/serialization/fb_serializer.h:298]: This should be another function in context_encoding.h

File: unit_test/fb_serializer_test.cc [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #define BOOST_TEST_DYN_LINK
 #ifdef STAND_ALONE
 #   define BOOST_TEST_MODULE Main
-#endif
+#endif
```
Comment (kumpera) [inline unit_test/fb_serializer_test.cc:4]: nit: line damage

File: rlclientlib/serialization/fb_serializer.h [hunk 4]
```diff
@@ -273,6 +275,59 @@ namespace reinforcement_learning { namespace logger {
...
+      } else {
+        return "IDENTITY";
+      }
+    }
+
+    void finalize()
+    {
+      auto event_offsets = _builder.CreateVector(_event_offsets);
+
+            const auto batch_metadata_offset = v2::CreateBatchMetadataDirect(_builder, getTextForEnum(_content_encoding));
```
Comment (kumpera) [inline rlclientlib/serialization/fb_serializer.h:311]: nit: bad identation. @jackgerrits don't we have a bot for this?

**Review by kumpera — COMMENTED** [review]
The PR lacks test for content encoding in serialized data.
Another concern is the code duplication in fb_collection_serializer that can be avoided by introducing another templatized type to deal with adding the header.

We need to make sure, and test for, that we can't do any encoding but IDENTITY with schema v1.

File: rlclientlib/serialization/fb_serializer.h (hunk 2 above)
Comment (cheng-tan) [reply rlclientlib/serialization/fb_serializer.h:298]: This will be entirely removed. I'm changing content_encoding to string instead of enum.

**Review by kumpera — APPROVED** [review]

**Pull request merged**